gems/aws-sdk-core/lib/aws-sdk-core/shared_credentials.rb

```diff
@@ -25,57 +26,96 @@ class SharedCredentials
     # You may access the resolved credentials through
     # `client.config.credentials`.
     #
-    # @option [String] :path Path to the shared file.  Defaults
+    # @option [String] :path Path to the shared file. Defaults
     #   to "#{Dir.home}/.aws/credentials".
     #
     # @option [String] :profile_name Defaults to 'default' or
     #   `ENV['AWS_PROFILE']`.
     #
+    # @option [Integer] :refresh_interval The duration (in seconds) before the
+    #   credentials are considered near expiration and should be refreshed.
+    #   Defaults to 10 minutes (600 seconds).
+    #
+    # @option [Boolean] :enable_refresh If true, the credentials will be automatically
+    #   refreshed when they are near expiration. Defaults to false.
+    #
     def initialize(options = {})
-      shared_config = Aws.shared_config
-      @path = options[:path]
-      @path ||= shared_config.credentials_path
-      @profile_name = options[:profile_name]
-      @profile_name ||= ENV['AWS_PROFILE']
-      @profile_name ||= shared_config.profile_name
-      if @path && @path == shared_config.credentials_path
-        @credentials = shared_config.credentials(profile: @profile_name)
-      else
-        config = SharedConfig.new(
-          credentials_path: @path,
-          profile_name: @profile_name
-        )
-        @credentials = config.credentials(profile: @profile_name)
-      end
+      @enable_refresh = options[:enable_refresh] || false # Refresh disabled by default
+      @refresh_interval = options[:refresh_interval] || 10 * 60 # Default refresh interval: 10 minutes
```

Comment:
I think the refresh_interval is necessary here because `expiration` isn't available from the credentials file, so the code below in `refresh` is using it to trigger when a refresh will happen.